src/brogue/Items.c

```diff
@@ -363,16 +362,14 @@ short chooseKind(const itemTable *theTable, short numKinds) {
 }
 
 // Places an item at (x,y) if provided or else a random location if they're 0. Inserts item into the floor list.
-item *placeItem(item *theItem, short x, short y) {
+item *placeItem(item *theItem, pos dest) {
     enum dungeonLayers layer;
     char theItemName[DCOLS], buf[DCOLS];
-    if (x <= 0 || y <= 0) {
-        pos loc;
-        randomMatchingLocation(&loc.x, &loc.y, FLOOR, NOTHING, -1);
-        theItem->loc = loc;
```

Comment:
I think this version of code was accidentally buggy (introduced by me earlier) it doesn't modify `x` and `y` even though they're used below in this function. Only a handful of items (maybe just the amulet?) get "invalid" `x` and `y` as args, so it would be rare.

With this fix, `dest` is now properly modified so things should work even in those edge cases.